cmake/modules/thirdparty/FindCSharp.cmake

```diff
@@ -33,16 +34,28 @@ if( NOT ${CSHARP_PLATFORM} MATCHES "x86|x64|anycpu|itanium" )
   message( FATAL_ERROR "The C# target platform '${CSHARP_PLATFORM}' is not valid. Please enter one of the following: x86, x64, anycpu, or itanium" )
 endif( )
 
-if( WIN32 )
-  find_package( DotNetFrameworkSdk )
-  if( NOT CSHARP_DOTNET_FOUND )
-    find_package( Mono )
-  endif( )
-else( UNIX )
+if( CSHARP_MONO )
   find_package( Mono )
-endif( )
+else()
+  find_package( Dotnet )
```

Comment:
what is the difference between the find_package(Dotnet) and find_package( DotNetFrameworkSdk ) ? Should we really keep both, or one can supersede the other one ?

Reply:
FindDotnet is looking for `dotnet.exe` and FindDotNetFrameworkSdk is looking for `csc.exe`. `dotnet.exe` is the modern version and covers Net5.0 and compatible earlier versions - like .NET Standard 2.0 and 2.1 etc. I believe that there if you have .NET Framework you would still need to look for `csc.exe` - so I have left it in. It is largely a legacy thing - since I don't think anyone does actually compile the bindings with .NET Framework.

Reply:
> It is largely a legacy thing - since I don't think anyone does actually compile the bindings with .NET Framework.

@szekerest @bjornharrtell opinons ? i'd say that if it is a legacy thing, we could remove support from that if that can simplify things

Reply:
In my opinion it is fine to drop .NET Framework. Since .NET 5 I cannot imagine any reason to look back regardless of platform etc.

Reply:
That looks like a quorum to drop .NET Framework support and csc.exe. Give me a nod and I will push a commit through to do that.

Reply:
In the absence of further comment - I have gone ahead and removed .NET Framework